server/util/db/db.go

```diff
@@ -234,6 +239,96 @@ func (m *mysqlDSNFormatter) Clone() dsnFormatter {
 	return &mysqlDSNFormatter{cfg: m.cfg.Clone()}
 }
 
+type postgresDSNFormatter struct {
+	host     string
+	port     string
+	user     string
+	password string
+	dbname   string
+	params   map[string]string
+}
+
+func newPostgresDSNFormatter(ac *AdvancedConfig) (*postgresDSNFormatter, error) {
+	endpoint, err := url.Parse(ac.Endpoint)
+	if err != nil {
+		return nil, status.FailedPreconditionErrorf("endpoint %s is not a valid URL: %s", ac.Endpoint, err)
+	}
+	params := make(map[string]string, 0)
+	if ac.Params != "" {
+		for _, param := range strings.Split(ac.Params, "&") {
+			ix := strings.Index(param, "&")
+			if ix < 0 {
+				return nil, status.FailedPreconditionErrorf("endpoint %s is not a valid URL: %s", ac.Endpoint, err)
+			}
+			key := param[:ix]
+			value := param[ix+1:]
+			params[key] = value
+		}
+	}
+	p := &postgresDSNFormatter{
+		host:     endpoint.Hostname(),
+		port:     endpoint.Port(),
+		user:     ac.Username,
+		password: ac.Password,
+		dbname:   ac.DBName,
+		params:   params,
+	}
+	p.AddParam("sslmode", "disable")
+	return p, nil
+}
+
+func (p *postgresDSNFormatter) AddParam(key, val string) {
+	p.params[key] = val
+}
+
+func (p *postgresDSNFormatter) String() string {
+	dsn := []string{}
+	if p.host != "" {
+		dsn = append(dsn, fmt.Sprintf("host = %s", p.host))
```

Comment:
It looks like the postgres driver supports the connection string in the same format as MySQL and sqlite: https://pkg.go.dev/github.com/jackc/pgx/v5/pgconn#ParseConfig

Would it be simpler if we stuck to this format instead of using something different for postgres?